#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_64bit.dpatch by Jose Carlos Medeiros <debian@psabs.com.br>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Avoid 32 bit counter overflow

###########
# http://www.isthe.com/chongo/src/webalizer-patch/

##################
# For very busy sites, 32 bit signed counters can overflow.  This
# is particularly when using webalizer to cover a long span of time.
# This patch converts a few values to be u_int64_t to avoid these
# numeric overflow problems.
#
##################

# Solaris users:
#
#    Solaris uses u_longlong_t instead of u_int64_t.  So when you build and install, try:
#
#	cd somepath/webalizer-2.01-10
#	rm -f config.cache
#	CFLAGS="$CFLAGS -Du_int64_t=u_longlong_t" ./configure [--optional_flags ...]
#	make all
#	make install
#
# See:
#	./configure --help
#
# for a list of optional flags.

@DPATCH@
Index: webalizer-2.01.10-vc/README.FIRST
===================================================================
--- webalizer-2.01.10-vc.orig/README.FIRST	2006-10-27 12:34:31.000000000 +0400
+++ webalizer-2.01.10-vc/README.FIRST	2006-10-27 12:34:32.000000000 +0400
@@ -231,3 +231,30 @@
 # seems to catch most of the longer values.
 ##################

+###################  SUMMARY 02_64bit  ##########################
+
+##################
+# For very busy sites, 32 bit signed counters can overflow.  This
+# is particularly when using webalizer to cover a long span of time.
+# This patch converts a few values to be u_int64_t to avoid these
+# numeric overflow problems.
+#
+# NOTE: While not required, it is recommened that you apply the
+#      patch-0.basic.ptch before you apply this patch.
+##################
+#Solaris users:
+
+#Solaris uses u_longlong_t instead of u_int64_t.  So when you build and install, try:
+#
+#       cd somepath/webalizer-2.01-10
+#       rm -f config.cache
+#       CFLAGS="$CFLAGS -Du_int64_t=u_longlong_t" ./configure [--optional_flags ...]
+#       make all
+#       make install
+#
+#See:
+#       ./configure --help
+#
+#for a list of optional flags.
+
+
Index: webalizer-2.01.10-vc/dns_resolv.c
===================================================================
--- webalizer-2.01.10-vc.orig/dns_resolv.c	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/dns_resolv.c	2006-10-27 12:34:32.000000000 +0400
@@ -178,7 +178,7 @@
    int       i;
    int       save_verbose=verbose;

-   u_long    listEntries = 0;
+   u_int64_t    listEntries = 0;

    struct sigaction sigPipeAction;
    struct stat dbStat;
@@ -338,7 +338,7 @@
    if (time_me || (verbose>1))
    {
       if (verbose<2 && time_me) printf("DNS: ");
-      printf("%lu %s ",listEntries, msg_addresses);
+      printf("%lld %s ",listEntries, msg_addresses);

       /* get processing time (end-start) */
       temp_time = (float)(end_time-start_time)/CLK_TCK;
Index: webalizer-2.01.10-vc/graphs.c
===================================================================
--- webalizer-2.01.10-vc.orig/graphs.c	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/graphs.c	2006-10-27 12:34:32.000000000 +0400
@@ -71,7 +71,7 @@
 FILE		*out;                      /* output file for PNG */
 char		maxvaltxt[32];             /* graph values        */
 float		percent;                   /* percent storage     */
-u_long		julday;                    /* julday value        */
+u_int64_t		julday;                    /* julday value        */

 struct pie_data { int x; int y;            /* line x,y            */
                   int mx; int my; };       /* midpoint x,y        */
@@ -88,19 +88,19 @@
 int year_graph6x(  char *fname,            /* file name use      */
                    char *title,            /* title for graph    */
                     int fmonth,            /* begin month number */
-                 u_long data1[12],         /* data1 (hits)       */
-                 u_long data2[12],         /* data2 (files)      */
-                 u_long data3[12],         /* data3 (sites)      */
+                 u_int64_t data1[12],         /* data1 (hits)       */
+                 u_int64_t data2[12],         /* data2 (files)      */
+                 u_int64_t data3[12],         /* data3 (sites)      */
                  double data4[12],         /* data4 (kbytes)     */
-                 u_long data5[12],         /* data5 (views)      */
-                 u_long data6[12])         /* data6 (visits)     */
+                 u_int64_t data5[12],         /* data5 (views)      */
+                 u_int64_t data6[12])         /* data6 (visits)     */
 {

    /* local variables */
    int i,j,x1,y1,x2;
    int s_mth;

-   u_long maxval=1;
+   u_int64_t maxval=1;
    double fmaxval=0.0;

    /* initalize the graph */
@@ -137,7 +137,7 @@
       if (data5[i] > maxval) maxval = data5[i];
    }
    if (maxval <= 0) maxval = 1;
-   sprintf(maxvaltxt, "%lu", maxval);
+   sprintf(maxvaltxt, "%lld", maxval);
    gdImageStringUp(im,gdFontSmall,8,26+(strlen(maxvaltxt)*6),maxvaltxt,black);

    if (graph_legend)                          /* print color coded legends? */
@@ -221,7 +221,7 @@
        if (data6[i] > maxval) maxval = data6[i];
    }
    if (maxval <= 0) maxval = 1;
-   sprintf(maxvaltxt, "%lu", maxval);
+   sprintf(maxvaltxt, "%lld", maxval);
    gdImageStringUp(im, gdFontSmall,493,26+(strlen(maxvaltxt)*6),
                    maxvaltxt, black);

@@ -299,17 +299,17 @@
                    char *title,            /* graph title        */
                     int month,             /* graph month        */
                     int year,              /* graph year         */
-                 u_long data1[31],         /* data1 (hits)       */
-                 u_long data2[31],         /* data2 (files)      */
-                 u_long data3[31],         /* data3 (sites)      */
+                 u_int64_t data1[31],         /* data1 (hits)       */
+                 u_int64_t data2[31],         /* data2 (files)      */
+                 u_int64_t data3[31],         /* data3 (sites)      */
                  double data4[31],         /* data4 (kbytes)     */
-                 u_long data5[31],         /* data5 (views)      */
-                 u_long data6[31])         /* data6 (visits)     */
+                 u_int64_t data5[31],         /* data5 (views)      */
+                 u_int64_t data6[31])         /* data6 (visits)     */
 {

    /* local variables */
    int i,j,s,x1,y1,x2;
-   u_long maxval=0;
+   u_int64_t maxval=0;
    double fmaxval=0.0;

    /* calc julian date for month */
@@ -354,7 +354,7 @@
        if (data5[i] > maxval) maxval = data5[i];
    }
    if (maxval <= 0) maxval = 1;
-   sprintf(maxvaltxt, "%lu", maxval);
+   sprintf(maxvaltxt, "%lld", maxval);
    gdImageStringUp(im, gdFontSmall,8,26+(strlen(maxvaltxt)*6),
                    maxvaltxt,black);

@@ -434,7 +434,7 @@
       if (data6[i]>maxval) maxval = data6[i];
    }
    if (maxval <= 0) maxval = 1;
-   sprintf(maxvaltxt, "%lu", maxval);
+   sprintf(maxvaltxt, "%lld", maxval);
    gdImageStringUp(im, gdFontSmall,8,180+(strlen(maxvaltxt)*6),
                    maxvaltxt, black);

@@ -502,14 +502,14 @@

 int day_graph3(  char *fname,
                  char *title,
-               u_long data1[24],
-               u_long data2[24],
-               u_long data3[24])
+               u_int64_t data1[24],
+               u_int64_t data2[24],
+               u_int64_t data3[24])
 {

    /* local variables */
    int i,j,s,x1,y1,x2;
-   u_long maxval=0;
+   u_int64_t maxval=0;

    /* initalize the graph */
    init_graph(title,512,256);
@@ -531,7 +531,7 @@
       if (data3[i] > maxval) maxval = data3[i];
    }
    if (maxval <= 0) maxval = 1;
-   sprintf(maxvaltxt, "%lu", maxval);
+   sprintf(maxvaltxt, "%lld", maxval);
    gdImageStringUp(im, gdFontSmall, 8, 26+(strlen(maxvaltxt)*6),
                    maxvaltxt, black);

@@ -607,8 +607,8 @@
 /*                                                               */
 /*****************************************************************/

-int pie_chart(char *fname, char *title, u_long t_val,
-              u_long data1[], char *legend[])
+int pie_chart(char *fname, char *title, u_int64_t t_val,
+              u_int64_t data1[], char *legend[])
 {
    int i,x,percent,y=47;
    double s_arc=0.0;
Index: webalizer-2.01.10-vc/graphs.h
===================================================================
--- webalizer-2.01.10-vc.orig/graphs.h	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/graphs.h	2006-10-27 12:34:32.000000000 +0400
@@ -1,11 +1,11 @@
 #ifndef _GRAPHS_H
 #define _GRAPHS_H

-extern int  month_graph6(char *, char *, int, int, u_long *,
-             u_long *, u_long *, double *, u_long *, u_long *);
+extern int  month_graph6(char *, char *, int, int, u_int64_t *,
+             u_int64_t *, u_int64_t *, double *, u_int64_t *, u_int64_t *);
 extern int  year_graph6x(char *, char *, int,
-             u_long *, u_long *, u_long *, double *, u_long *, u_long *);
-extern int  day_graph3(char *, char *, u_long *, u_long *, u_long *);
-extern int  pie_chart(char *, char *, u_long, u_long *, char **);
+             u_int64_t *, u_int64_t *, u_int64_t *, double *, u_int64_t *, u_int64_t *);
+extern int  day_graph3(char *, char *, u_int64_t *, u_int64_t *, u_int64_t *);
+extern int  pie_chart(char *, char *, u_int64_t, u_int64_t *, char **);

 #endif  /* _GRAPHS_H */
Index: webalizer-2.01.10-vc/hashtab.c
===================================================================
--- webalizer-2.01.10-vc.orig/hashtab.c	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/hashtab.c	2006-10-27 12:34:32.000000000 +0400
@@ -79,7 +79,7 @@
 void     update_entry(char *);                /* update entry/exit        */
 void     update_exit(char *);                 /* page totals              */

-u_long   hash(char *);                        /* hash function            */
+u_int64_t   hash(char *);                        /* hash function            */

 /* local data */

@@ -154,12 +154,12 @@

 int put_hnode( char     *str,   /* Hostname  */
                int       type,  /* obj type  */
-               u_long    count, /* hit count */
-               u_long    file,  /* File flag */
+               u_int64_t    count, /* hit count */
+               u_int64_t    file,  /* File flag */
                double    xfer,  /* xfer size */
-               u_long   *ctr,   /* counter   */
-               u_long    visit, /* visits    */
-               u_long    tstamp,/* timestamp */
+               u_int64_t   *ctr,   /* counter   */
+               u_int64_t    visit, /* visits    */
+               u_int64_t    tstamp,/* timestamp */
                char     *lasturl, /* lasturl */
                HNODEPTR *htab)  /* ptr>next  */
 {
@@ -341,8 +341,8 @@
 /* PUT_UNODE - insert/update URL node        */
 /*********************************************/

-int put_unode(char *str, int type, u_long count, double xfer,
-              u_long *ctr, u_long entry, u_long exit, UNODEPTR *htab)
+int put_unode(char *str, int type, u_int64_t count, double xfer,
+              u_int64_t *ctr, u_int64_t entry, u_int64_t exit, UNODEPTR *htab)
 {
    UNODEPTR cptr,nptr;

@@ -468,7 +468,7 @@
 /* PUT_RNODE - insert/update referrer node   */
 /*********************************************/

-int put_rnode(char *str, int type, u_long count, u_long *ctr, RNODEPTR *htab)
+int put_rnode(char *str, int type, u_int64_t count, u_int64_t *ctr, RNODEPTR *htab)
 {
    RNODEPTR cptr,nptr;

@@ -587,7 +587,7 @@
 /* PUT_ANODE - insert/update user agent node */
 /*********************************************/

-int put_anode(char *str, int type, u_long count, u_long *ctr, ANODEPTR *htab)
+int put_anode(char *str, int type, u_int64_t count, u_int64_t *ctr, ANODEPTR *htab)
 {
    ANODEPTR cptr,nptr;

@@ -702,7 +702,7 @@
 /* PUT_SNODE - insert/update search str node */
 /*********************************************/

-int put_snode(char *str, u_long count, SNODEPTR *htab)
+int put_snode(char *str, u_int64_t count, SNODEPTR *htab)
 {
    SNODEPTR cptr,nptr;

@@ -810,12 +810,12 @@

 int put_inode( char     *str,   /* ident str */
                int       type,  /* obj type  */
-               u_long    count, /* hit count */
-               u_long    file,  /* File flag */
+               u_int64_t    count, /* hit count */
+               u_int64_t    file,  /* File flag */
                double    xfer,  /* xfer size */
-               u_long   *ctr,   /* counter   */
-               u_long    visit, /* visits    */
-               u_long    tstamp,/* timestamp */
+               u_int64_t   *ctr,   /* counter   */
+               u_int64_t    visit, /* visits    */
+               u_int64_t    tstamp,/* timestamp */
                INODEPTR *htab)  /* hashtable */
 {
    INODEPTR cptr,nptr;
@@ -1047,9 +1047,9 @@
 /* HASH - return hash value for string       */
 /*********************************************/

-u_long hash(char *str)
+u_int64_t hash(char *str)
 {
-   u_long hashval;
+   u_int64_t hashval;
    for (hashval = 0; *str != '\0'; str++)
       hashval = *str + 31 * hashval;
    return hashval % MAXHASH;
@@ -1133,7 +1133,7 @@
 /* MONTH_UPDATE_EXIT  - eom exit page update */
 /*********************************************/

-void month_update_exit(u_long tstamp)
+void month_update_exit(u_int64_t tstamp)
 {
    HNODEPTR nptr;
    int i;
@@ -1157,10 +1157,10 @@
 /* TOT_VISIT - calculate total visits        */
 /*********************************************/

-u_long tot_visit(HNODEPTR *list)
+u_int64_t tot_visit(HNODEPTR *list)
 {
    HNODEPTR   hptr;
-   u_long     tot=0;
+   u_int64_t     tot=0;
    int        i;

    for (i=0;i<MAXHASH;i++)
Index: webalizer-2.01.10-vc/hashtab.h
===================================================================
--- webalizer-2.01.10-vc.orig/hashtab.h	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/hashtab.h	2006-10-27 12:34:32.000000000 +0400
@@ -25,43 +25,43 @@

 struct hnode {  char *string;              /* host hash table structure    */
                  int flag;
-              u_long count;
-              u_long files;
-              u_long visit;                /* visit information            */
-              u_long tstamp;
+              u_int64_t count;
+              u_int64_t files;
+              u_int64_t visit;                /* visit information            */
+              u_int64_t tstamp;
                 char *lasturl;
               double xfer;
               struct hnode *next; };

 struct unode {  char *string;              /* url hash table structure     */
                  int flag;                 /* Object type (REG, HIDE, GRP) */
-              u_long count;                /* requests counter             */
-              u_long files;                /* files counter                */
-              u_long entry;                /* entry page counter           */
-              u_long exit;                 /* exit page counter            */
+              u_int64_t count;                /* requests counter             */
+              u_int64_t files;                /* files counter                */
+              u_int64_t entry;                /* entry page counter           */
+              u_int64_t exit;                 /* exit page counter            */
               double xfer;                 /* xfer size in bytes           */
               struct unode *next; };       /* pointer to next node         */

 struct rnode {  char *string;              /* referrer hash table struct   */
                  int flag;
-              u_long count;
+              u_int64_t count;
               struct rnode *next; };

 struct anode {  char *string;
                  int flag;
-              u_long count;
+              u_int64_t count;
               struct anode *next; };

 struct snode {  char *string;                 /* search string struct      */
-              u_long count;
+              u_int64_t count;
               struct snode *next; };

 struct inode {  char *string;                 /* host hash table struct    */
                  int flag;
-              u_long count;
-              u_long files;
-              u_long visit;
-              u_long tstamp;
+              u_int64_t count;
+              u_int64_t files;
+              u_int64_t visit;
+              u_int64_t tstamp;
               double xfer;
               struct inode *next; };

@@ -76,15 +76,15 @@
 extern DNODEPTR host_table[MAXHASH];          /* DNS resolver table        */
 #endif

-extern int    put_hnode(char *, int, u_long, u_long, double, u_long *,
-                        u_long, u_long, char *, HNODEPTR *);
-extern int    put_unode(char *, int, u_long, double, u_long *,
-                        u_long, u_long, UNODEPTR *);
-extern int    put_inode(char *, int, u_long, u_long, double, u_long *,
-                        u_long, u_long, INODEPTR *);
-extern int    put_rnode(char *, int, u_long, u_long *, RNODEPTR *);
-extern int    put_anode(char *, int, u_long, u_long *, ANODEPTR *);
-extern int    put_snode(char *, u_long, SNODEPTR *);
+extern int    put_hnode(char *, int, u_int64_t, u_int64_t, double, u_int64_t *,
+                        u_int64_t, u_int64_t, char *, HNODEPTR *);
+extern int    put_unode(char *, int, u_int64_t, double, u_int64_t *,
+                        u_int64_t, u_int64_t, UNODEPTR *);
+extern int    put_inode(char *, int, u_int64_t, u_int64_t, double, u_int64_t *,
+                        u_int64_t, u_int64_t, INODEPTR *);
+extern int    put_rnode(char *, int, u_int64_t, u_int64_t *, RNODEPTR *);
+extern int    put_anode(char *, int, u_int64_t, u_int64_t *, ANODEPTR *);
+extern int    put_snode(char *, u_int64_t, SNODEPTR *);

 #ifdef USE_DNS
 extern int    put_dnode(char *, struct in_addr *, DNODEPTR *);
@@ -99,8 +99,8 @@
 extern void   del_slist(SNODEPTR *);          /* delete host htab          */
 extern void   del_ilist(INODEPTR *);          /* delete host htab          */

-extern void   month_update_exit(u_long);
-extern u_long tot_visit(HNODEPTR *);
+extern void   month_update_exit(u_int64_t);
+extern u_int64_t tot_visit(HNODEPTR *);
 extern char   *find_url(char *);

 #endif  /* _HASHTAB_H */
Index: webalizer-2.01.10-vc/output.c
===================================================================
--- webalizer-2.01.10-vc.orig/output.c	2006-10-27 12:34:31.000000000 +0400
+++ webalizer-2.01.10-vc/output.c	2006-10-27 12:34:32.000000000 +0400
@@ -82,12 +82,12 @@
 void    top_ctry_table();                           /* top n countries ""  */
 void    top_search_table();                         /* top n search strs   */
 void    top_users_table();                          /* top n ident table   */
-u_long  load_url_array(  UNODEPTR *);               /* load URL array      */
-u_long  load_site_array( HNODEPTR *);               /* load Site array     */
-u_long  load_ref_array(  RNODEPTR *);               /* load Refs array     */
-u_long  load_agent_array(ANODEPTR *);               /* load Agents array   */
-u_long  load_srch_array( SNODEPTR *);               /* load srch str array */
-u_long  load_ident_array(INODEPTR *);               /* load ident array    */
+u_int64_t  load_url_array(  UNODEPTR *);               /* load URL array      */
+u_int64_t  load_site_array( HNODEPTR *);               /* load Site array     */
+u_int64_t  load_ref_array(  RNODEPTR *);               /* load Refs array     */
+u_int64_t  load_agent_array(ANODEPTR *);               /* load Agents array   */
+u_int64_t  load_srch_array( SNODEPTR *);               /* load srch str array */
+u_int64_t  load_ident_array(INODEPTR *);               /* load ident array    */
 int	qs_url_cmph( const void*, const void*);     /* compare by hits     */
 int	qs_url_cmpk( const void*, const void*);     /* compare by kbytes   */
 int	qs_url_cmpn( const void*, const void*);     /* compare by entrys   */
@@ -100,12 +100,12 @@
 int     qs_ident_cmph(const void*, const void*);    /* compare by hits     */
 int     qs_ident_cmpk(const void*, const void*);    /* compare by kbytes   */

-int     all_sites_page(u_long, u_long);             /* output site page    */
-int     all_urls_page(u_long, u_long);              /* output urls page    */
-int     all_refs_page(u_long, u_long);              /* output refs page    */
-int     all_agents_page(u_long, u_long);            /* output agents page  */
-int     all_search_page(u_long, u_long);            /* output search page  */
-int     all_users_page(u_long, u_long);             /* output ident page   */
+int     all_sites_page(u_int64_t, u_int64_t);             /* output site page    */
+int     all_urls_page(u_int64_t, u_int64_t);              /* output urls page    */
+int     all_refs_page(u_int64_t, u_int64_t);              /* output refs page    */
+int     all_agents_page(u_int64_t, u_int64_t);            /* output agents page  */
+int     all_search_page(u_int64_t, u_int64_t);            /* output search page  */
+int     all_users_page(u_int64_t, u_int64_t);             /* output ident page   */
 void    dump_all_sites();                           /* dump sites tab file */
 void    dump_all_urls();                            /* dump urls tab file  */
 void    dump_all_refs();                            /* dump refs tab file  */
@@ -136,7 +136,7 @@
 ANODEPTR *a_array      = NULL;                /* user agents              */
 SNODEPTR *s_array      = NULL;                /* search strings           */
 INODEPTR *i_array      = NULL;                /* ident strings (username) */
-u_long   a_ctr         = 0;                   /* counter for sort array   */
+u_int64_t   a_ctr         = 0;                   /* counter for sort array   */

 FILE     *out_fp;

@@ -508,7 +508,7 @@
 void month_total_table()
 {
    int i,days_in_month;
-   u_long max_files=0,max_hits=0,max_visits=0,max_pages=0;
+   u_int64_t max_files=0,max_hits=0,max_visits=0,max_pages=0;
    double max_xfer=0.0;

    days_in_month=(l_day-f_day)+1;
@@ -528,19 +528,19 @@
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    /* Total Hits */
    fprintf(out_fp,"<TR><TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%llu</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_th,t_hit);
    /* Total Files */
    fprintf(out_fp,"<TR><TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_tf,t_file);
    /* Total Pages */
    fprintf(out_fp,"<TR><TD WIDTH=380><FONT SIZE=\"-1\">%s %s</FONT></TD>\n"  \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_h_total, msg_h_pages, t_page);
    /* Total Visits */
    fprintf(out_fp,"<TR><TD WIDTH=380><FONT SIZE=\"-1\">%s %s</FONT></TD>\n"  \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_h_total, msg_h_visits, t_visit);
    /* Total XFer */
    fprintf(out_fp,"<TR><TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"     \
@@ -551,30 +551,30 @@
    /* Unique Sites */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"                     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_us,t_site);
    /* Unique URL's */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"                     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_uu,t_url);
    /* Unique Referrers */
    if (t_ref != 0)
    fprintf(out_fp,"<TR>"                                                     \
       "<TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"                     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_ur,t_ref);
    /* Unique Usernames */
    if (t_user != 0)
    fprintf(out_fp,"<TR>"                                                     \
       "<TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"                     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_ui,t_user);
    /* Unique Agents */
    if (t_agent != 0)
    fprintf(out_fp,"<TR>"                                                     \
       "<TD WIDTH=380><FONT SIZE=\"-1\">%s</FONT></TD>\n"                     \
-      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"               \
+      "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"               \
       "</FONT></TD></TR>\n",msg_mtot_ua,t_agent);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    /**********************************************/
@@ -590,32 +590,32 @@
    /* Max/Avg Hits per Hour */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD><FONT SIZE=\"-1\">%s</FONT></TD>\n"                               \
-      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
-      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lu</B>"                    \
+      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
+      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%llu</B>"                    \
       "</FONT></TD></TR>\n",msg_mtot_mhh, t_hit/(24*days_in_month),mh_hit);
    /* Max/Avg Hits per Day */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD><FONT SIZE=\"-1\">%s</FONT></TD>\n"                               \
-      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
-      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lu</B>"                    \
+      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
+      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%llu</B>"                    \
       "</FONT></TD></TR>\n",msg_mtot_mhd, t_hit/days_in_month, max_hits);
    /* Max/Avg Files per Day */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD><FONT SIZE=\"-1\">%s</FONT></TD>\n"                               \
-      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
-      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lu</B>"                    \
+      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
+      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lld</B>"                    \
       "</FONT></TD></TR>\n",msg_mtot_mfd, t_file/days_in_month,max_files);
    /* Max/Avg Pages per Day */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD><FONT SIZE=\"-1\">%s</FONT></TD>\n"                               \
-      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
-      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lu</B>"                    \
+      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
+      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lld</B>"                    \
       "</FONT></TD></TR>\n",msg_mtot_mpd, t_page/days_in_month,max_pages);
    /* Max/Avg Visits per Day */
    fprintf(out_fp,"<TR>"                                                     \
       "<TD><FONT SIZE=\"-1\">%s</FONT></TD>\n"                               \
-      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
-      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lu</B>"                    \
+      "<TD ALIGN=right WIDTH=65><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
+      "<TD WIDTH=65 ALIGN=right><FONT SIZE=-1><B>%lld</B>"                    \
       "</FONT></TD></TR>\n",msg_mtot_mvd, t_visit/days_in_month,max_visits);
    /* Max/Avg KBytes per Day */
    fprintf(out_fp,"<TR>"                                                     \
@@ -634,7 +634,7 @@
    {
       if (response[i].count != 0)
          fprintf(out_fp,"<TR><TD><FONT SIZE=\"-1\">%s</FONT></TD>\n"         \
-            "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lu</B>"         \
+            "<TD ALIGN=right COLSPAN=2><FONT SIZE=\"-1\"><B>%lld</B>"         \
             "</FONT></TD></TR>\n",
             response[i].desc, response[i].count);
    }
@@ -693,23 +693,23 @@
       fprintf(out_fp,"<TR><TD ALIGN=center>"                                 \
               "<FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n", i+1);
       fprintf(out_fp,"<TD ALIGN=right>"                                      \
-              "<FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"                   \
+              "<FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"                   \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
               tm_hit[i],PCENT(tm_hit[i],t_hit));
       fprintf(out_fp,"<TD ALIGN=right>"                                      \
-              "<FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"                   \
+              "<FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"                   \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
               tm_file[i],PCENT(tm_file[i],t_file));
       fprintf(out_fp,"<TD ALIGN=right>"                                      \
-              "<FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"                   \
+              "<FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"                   \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
               tm_page[i],PCENT(tm_page[i],t_page));
       fprintf(out_fp,"<TD ALIGN=right>"                                      \
-              "<FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"                   \
+              "<FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"                   \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
               tm_visit[i],PCENT(tm_visit[i],t_visit));
       fprintf(out_fp,"<TD ALIGN=right>"                                      \
-              "<FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"                   \
+              "<FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"                   \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
               tm_site[i],PCENT(tm_site[i],t_site));
       fprintf(out_fp,"<TD ALIGN=right>"                                      \
@@ -729,7 +729,7 @@
 void hourly_total_table()
 {
    int i,days_in_month;
-   u_long avg_file=0;
+   u_int64_t avg_file=0;
    double avg_xfer=0.0;

    days_in_month=(l_day-f_day)+1;
@@ -783,20 +783,20 @@
       fprintf(out_fp,"<TR><TD ALIGN=center>"                          \
          "<FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n",i);
       fprintf(out_fp,
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
          "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
          th_hit[i]/days_in_month,th_hit[i],
          PCENT(th_hit[i],t_hit));
       fprintf(out_fp,
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
          "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
          th_file[i]/days_in_month,th_file[i],
          PCENT(th_file[i],t_file));
       fprintf(out_fp,
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
          "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n",
          th_page[i]/days_in_month,th_page[i],
          PCENT(th_page[i],t_page));
@@ -819,7 +819,7 @@

 void top_sites_table(int flag)
 {
-   u_long cnt=0, h_reg=0, h_grp=0, h_hid=0, tot_num;
+   u_int64_t cnt=0, h_reg=0, h_grp=0, h_hid=0, tot_num;
    int i;
    HNODEPTR hptr, *pointer;

@@ -846,11 +846,11 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    if (flag) fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=10>" \
-           "%s %lu %s %lu %s %s %s</TH></TR>\n",
+           "%s %lld %s %lld %s %s %s</TH></TR>\n",
            GREY, msg_top_top,tot_num,msg_top_of,
            t_site,msg_top_s,msg_h_by,msg_h_xfer);
    else      fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=10>" \
-           "%s %lu %s %lu %s</TH></TR>\n",
+           "%s %lld %s %lld %s</TH></TR>\n",
            GREY,msg_top_top, tot_num, msg_top_of, t_site, msg_top_s);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=center>"                    \
@@ -880,13 +880,13 @@

          fprintf(out_fp,
               "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n"  \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
               "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%.0f</B></FONT></TD>\n" \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
               "<TD ALIGN=left NOWRAP><FONT SIZE=\"-1\">",
               i+1,hptr->count,
@@ -930,7 +930,7 @@
 /* ALL_SITES_PAGE - HTML page of all sites   */
 /*********************************************/

-int all_sites_page(u_long h_reg, u_long h_grp)
+int all_sites_page(u_int64_t h_reg, u_int64_t h_grp)
 {
    HNODEPTR hptr, *pointer;
    char     site_fname[256], buffer[256];
@@ -961,7 +961,7 @@
       if (hptr->flag == OBJ_GRP)
       {
          fprintf(out_fp,
-         "%-8lu %6.02f%%  %8lu %6.02f%%  %8.0f %6.02f%%  %8lu %6.02f%%  %s\n",
+         "%-8llu %6.02f%%  %8llu %6.02f%%  %8.0f %6.02f%%  %8llu %6.02f%%  %s\n",
             hptr->count,
             (t_hit==0)?0:((float)hptr->count/t_hit)*100.0,hptr->files,
             (t_file==0)?0:((float)hptr->files/t_file)*100.0,hptr->xfer/1024,
@@ -982,7 +982,7 @@
       if (hptr->flag == OBJ_REG)
       {
          fprintf(out_fp,
-         "%-8lu %6.02f%%  %8lu %6.02f%%  %8.0f %6.02f%%  %8lu %6.02f%%  %s\n",
+         "%-8llu %6.02f%%  %8llu %6.02f%%  %8.0f %6.02f%%  %8llu %6.02f%%  %s\n",
             hptr->count,
             (t_hit==0)?0:((float)hptr->count/t_hit)*100.0,hptr->files,
             (t_file==0)?0:((float)hptr->files/t_file)*100.0,hptr->xfer/1024,
@@ -1005,7 +1005,7 @@

 void top_urls_table(int flag)
 {
-   u_long cnt=0,u_reg=0,u_grp=0,u_hid=0, tot_num;
+   u_int64_t cnt=0,u_reg=0,u_grp=0,u_hid=0, tot_num;
    int i;
    UNODEPTR uptr, *pointer;

@@ -1031,11 +1031,11 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    if (flag) fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=6>"  \
-           "%s %lu %s %lu %s %s %s</TH></TR>\n",
+           "%s %lld %s %lld %s %s %s</TH></TR>\n",
            GREY,msg_top_top,tot_num,msg_top_of,
            t_url,msg_top_u,msg_h_by,msg_h_xfer);
    else fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=6>"   \
-           "%s %lu %s %lu %s</TH></TR>\n",
+           "%s %lld %s %lld %s</TH></TR>\n",
            GREY,msg_top_top,tot_num,msg_top_of,t_url,msg_top_u);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=center>"                  \
@@ -1064,7 +1064,7 @@

          fprintf(out_fp,
             "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n" \
-            "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+            "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
             "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%.0f</B></FONT></TD>\n"\
             "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
@@ -1135,7 +1135,7 @@
 /* ALL_URLS_PAGE - HTML page of all urls     */
 /*********************************************/

-int all_urls_page(u_long u_reg, u_long u_grp)
+int all_urls_page(u_int64_t u_reg, u_int64_t u_grp)
 {
    UNODEPTR uptr, *pointer;
    char     url_fname[256], buffer[256];
@@ -1165,7 +1165,7 @@
       uptr=*pointer++;
       if (uptr->flag == OBJ_GRP)
       {
-         fprintf(out_fp,"%-8lu %6.02f%%  %8.0f %6.02f%%  %s\n",
+         fprintf(out_fp,"%-8llu %6.02f%%  %8.0f %6.02f%%  %s\n",
             uptr->count,
             (t_hit==0)?0:((float)uptr->count/t_hit)*100.0,
             uptr->xfer/1024,
@@ -1184,7 +1184,7 @@
       uptr=*pointer++;
       if (uptr->flag == OBJ_REG)
       {
-         fprintf(out_fp,"%-8lu %6.02f%%  %8.0f %6.02f%%  %s\n",
+         fprintf(out_fp,"%-8llu %6.02f%%  %8.0f %6.02f%%  %s\n",
             uptr->count,
             (t_hit==0)?0:((float)uptr->count/t_hit)*100.0,
             uptr->xfer/1024,
@@ -1206,8 +1206,8 @@

 void top_entry_table(int flag)
 {
-   u_long cnt=0, u_entry=0, u_exit=0, tot_num;
-   u_long t_entry=0, t_exit=0;
+   u_int64_t cnt=0, u_entry=0, u_exit=0, tot_num;
+   u_int64_t t_entry=0, t_exit=0;
    int i;
    UNODEPTR uptr, *pointer;

@@ -1216,10 +1216,10 @@
    {
       if ( (int)((UNODEPTR)(*pointer)->flag) == OBJ_REG )
       {
-         if ( (u_long)((UNODEPTR)(*pointer)->entry) )
-            {  u_entry++; t_entry+=(u_long)((UNODEPTR)(*pointer)->entry); }
-         if ( (u_long)((UNODEPTR)(*pointer)->exit)  )
-            { u_exit++;   t_exit +=(u_long)((UNODEPTR)(*pointer)->exit);  }
+         if ( (u_int64_t)(((UNODEPTR)(*pointer))->entry) )
+            {  u_entry++; t_entry+=(u_int64_t)(((UNODEPTR)(*pointer))->entry); }
+         if ( (u_int64_t)(((UNODEPTR)(*pointer))->exit)  )
+            { u_exit++;   t_exit +=(u_int64_t)(((UNODEPTR)(*pointer))->exit);  }
       }
       pointer++;
    }
@@ -1238,7 +1238,7 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=6>"        \
-           "%s %lu %s %lu %s</TH></TR>\n",
+           "%s %lld %s %lld %s</TH></TR>\n",
            GREY,msg_top_top,tot_num,msg_top_of,
            (flag)?u_exit:u_entry,(flag)?msg_top_ex:msg_top_en);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
@@ -1265,9 +1265,9 @@
          fprintf(out_fp,"<TR>\n");
          fprintf(out_fp,
              "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n" \
-             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
              "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
-             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
              "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
              "<TD ALIGN=left NOWRAP><FONT SIZE=\"-1\">",
              i+1,uptr->count,
@@ -1308,7 +1308,7 @@

 void top_refs_table()
 {
-   u_long cnt=0, r_reg=0, r_grp=0, r_hid=0, tot_num;
+   u_int64_t cnt=0, r_reg=0, r_grp=0, r_hid=0, tot_num;
    int i;
    RNODEPTR rptr, *pointer;

@@ -1334,7 +1334,7 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=4>"         \
-           "%s %lu %s %lu %s</TH></TR>\n",
+           "%s %lld %s %lld %s</TH></TR>\n",
            GREY, msg_top_top, tot_num, msg_top_of, t_ref, msg_top_r);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=center>"                   \
@@ -1361,7 +1361,7 @@

          fprintf(out_fp,
              "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n"  \
-             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
              "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
              "<TD ALIGN=left NOWRAP><FONT SIZE=\"-1\">",
              i+1,rptr->count,
@@ -1414,7 +1414,7 @@
 /* ALL_REFS_PAGE - HTML page of all refs     */
 /*********************************************/

-int all_refs_page(u_long r_reg, u_long r_grp)
+int all_refs_page(u_int64_t r_reg, u_int64_t r_grp)
 {
    RNODEPTR rptr, *pointer;
    char     ref_fname[256], buffer[256];
@@ -1442,7 +1442,7 @@
       rptr=*pointer++;
       if (rptr->flag == OBJ_GRP)
       {
-         fprintf(out_fp,"%-8lu %6.02f%%  %s\n",
+         fprintf(out_fp,"%-8llu %6.02f%%  %s\n",
             rptr->count,
             (t_hit==0)?0:((float)rptr->count/t_hit)*100.0,
             rptr->string);
@@ -1458,7 +1458,7 @@
       rptr=*pointer++;
       if (rptr->flag == OBJ_REG)
       {
-         fprintf(out_fp,"%-8lu %6.02f%%  %s\n",
+         fprintf(out_fp,"%-8llu %6.02f%%  %s\n",
             rptr->count,
             (t_hit==0)?0:((float)rptr->count/t_hit)*100.0,
             rptr->string);
@@ -1478,7 +1478,7 @@

 void top_agents_table()
 {
-   u_long cnt, a_reg=0, a_grp=0, a_hid=0, tot_num;
+   u_int64_t cnt, a_reg=0, a_grp=0, a_hid=0, tot_num;
    int i;
    ANODEPTR aptr, *pointer;

@@ -1504,7 +1504,7 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=4>"        \
-          "%s %lu %s %lu %s</TH></TR>\n",
+          "%s %lld %s %lld %s</TH></TR>\n",
           GREY, msg_top_top, tot_num, msg_top_of, t_agent, msg_top_a);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=center>"                  \
@@ -1531,7 +1531,7 @@

          fprintf(out_fp,
              "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n" \
-             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+             "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
              "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
              "<TD ALIGN=left NOWRAP><FONT SIZE=\"-1\">",
              i+1,aptr->count,
@@ -1566,7 +1566,7 @@
 /* ALL_AGENTS_PAGE - HTML user agent page    */
 /*********************************************/

-int all_agents_page(u_long a_reg, u_long a_grp)
+int all_agents_page(u_int64_t a_reg, u_int64_t a_grp)
 {
    ANODEPTR aptr, *pointer;
    char     agent_fname[256], buffer[256];
@@ -1594,7 +1594,7 @@
       aptr=*pointer++;
       if (aptr->flag == OBJ_GRP)
       {
-         fprintf(out_fp,"%-8lu %6.02f%%  %s\n",
+         fprintf(out_fp,"%-8llu %6.02f%%  %s\n",
              aptr->count,
              (t_hit==0)?0:((float)aptr->count/t_hit)*100.0,
              aptr->string);
@@ -1610,7 +1610,7 @@
       aptr=*pointer++;
       if (aptr->flag == OBJ_REG)
       {
-         fprintf(out_fp,"%-8lu %6.02f%%  %s\n",
+         fprintf(out_fp,"%-8llu %6.02f%%  %s\n",
              aptr->count,
              (t_hit==0)?0:((float)aptr->count/t_hit)*100.0,
              aptr->string);
@@ -1630,7 +1630,7 @@

 void top_search_table()
 {
-   u_long   cnt,t_val=0, tot_num;
+   u_int64_t   cnt,t_val=0, tot_num;
    int      i;
    SNODEPTR sptr, *pointer;

@@ -1639,7 +1639,7 @@
    cnt=tot_num=a_ctr; pointer=s_array;
    while(cnt--)
    {
-      t_val+=(u_long)((SNODEPTR)(*pointer)->count);
+      t_val+=(u_int64_t)(((SNODEPTR)(*pointer))->count);
       pointer++;
    }

@@ -1649,7 +1649,7 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=4>"        \
-          "%s %lu %s %lu %s</TH></TR>\n",
+          "%s %lld %s %lld %s</TH></TR>\n",
           GREY, msg_top_top, tot_num, msg_top_of, a_ctr, msg_top_sr);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=center>"                  \
@@ -1670,7 +1670,7 @@
       fprintf(out_fp,
          "<TR>\n"                                                     \
          "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n" \
-         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+         "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
          "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
          "<TD ALIGN=left NOWRAP><FONT SIZE=\"-1\">",
          i+1,sptr->count,
@@ -1699,7 +1699,7 @@
 /* ALL_SEARCH_PAGE - HTML for search strings */
 /*********************************************/

-int all_search_page(u_long tot_num, u_long t_val)
+int all_search_page(u_int64_t tot_num, u_int64_t t_val)
 {
    SNODEPTR sptr, *pointer;
    char     search_fname[256], buffer[256];
@@ -1725,7 +1725,7 @@
    while(tot_num)
    {
       sptr=*pointer++;
-      fprintf(out_fp,"%-8lu %6.02f%%  %s\n",
+      fprintf(out_fp,"%-8llu %6.02f%%  %s\n",
          sptr->count,
          (t_val==0)?0:((float)sptr->count/t_val)*100.0,
          sptr->string);
@@ -1743,7 +1743,7 @@

 void top_users_table()
 {
-   u_long cnt=0, i_reg=0, i_grp=0, i_hid=0, tot_num;
+   u_int64_t cnt=0, i_reg=0, i_grp=0, i_hid=0, tot_num;
    int i;
    INODEPTR iptr, *pointer;

@@ -1768,7 +1768,7 @@
    fprintf(out_fp,"<TABLE WIDTH=510 BORDER=2 CELLSPACING=1 CELLPADDING=1>\n");
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=CENTER COLSPAN=10>" \
-           "%s %lu %s %lu %s</TH></TR>\n",
+           "%s %lld %s %lld %s</TH></TR>\n",
            GREY,msg_top_top, tot_num, msg_top_of, t_user, msg_top_i);
    fprintf(out_fp,"<TR><TH HEIGHT=4></TH></TR>\n");
    fprintf(out_fp,"<TR><TH BGCOLOR=\"%s\" ALIGN=center>"                   \
@@ -1798,13 +1798,13 @@

          fprintf(out_fp,
               "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n"  \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
               "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%.0f</B></FONT></TD>\n" \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n"  \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n"  \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"    \
               "<TD ALIGN=left NOWRAP><FONT SIZE=\"-1\">",
               i+1,iptr->count,
@@ -1843,7 +1843,7 @@
 /* ALL_USERS_PAGE - HTML of all usernames    */
 /*********************************************/

-int all_users_page(u_long i_reg, u_long i_grp)
+int all_users_page(u_int64_t i_reg, u_int64_t i_grp)
 {
    INODEPTR iptr, *pointer;
    char     user_fname[256], buffer[256];
@@ -1874,7 +1874,7 @@
       if (iptr->flag == OBJ_GRP)
       {
          fprintf(out_fp,
-         "%-8lu %6.02f%%  %8lu %6.02f%%  %8.0f %6.02f%%  %8lu %6.02f%%  %s\n",
+         "%-8llu %6.02f%%  %8llu %6.02f%%  %8.0f %6.02f%%  %8llu %6.02f%%  %s\n",
             iptr->count,
             (t_hit==0)?0:((float)iptr->count/t_hit)*100.0,iptr->files,
             (t_file==0)?0:((float)iptr->files/t_file)*100.0,iptr->xfer/1024,
@@ -1895,7 +1895,7 @@
       if (iptr->flag == OBJ_REG)
       {
          fprintf(out_fp,
-         "%-8lu %6.02f%%  %8lu %6.02f%%  %8.0f %6.02f%%  %8lu %6.02f%%  %s\n",
+         "%-8llu %6.02f%%  %8llu %6.02f%%  %8.0f %6.02f%%  %8llu %6.02f%%  %s\n",
             iptr->count,
             (t_hit==0)?0:((float)iptr->count/t_hit)*100.0,iptr->files,
             (t_file==0)?0:((float)iptr->files/t_file)*100.0,iptr->xfer/1024,
@@ -1920,10 +1920,10 @@
 {
    int i,j,x,tot_num=0,tot_ctry=0;
    int ctry_fnd;
-   u_long idx;
+   u_int64_t idx;
    HNODEPTR hptr;
    char *domain;
-   u_long pie_data[10];
+   u_int64_t pie_data[10];
    char   *pie_legend[10];
    char   pie_title[48];
    char   pie_fname[48];
@@ -2040,9 +2040,9 @@
       if (top_ctrys[i]->count!=0)
       fprintf(out_fp,"<TR>"                                                \
               "<TD ALIGN=center><FONT SIZE=\"-1\"><B>%d</B></FONT></TD>\n" \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
-              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lu</B></FONT></TD>\n" \
+              "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%lld</B></FONT></TD>\n" \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
               "<TD ALIGN=right><FONT SIZE=\"-1\"><B>%.0f</B></FONT></TD>\n" \
               "<TD ALIGN=right><FONT SIZE=\"-2\">%3.02f%%</FONT></TD>\n"   \
@@ -2068,7 +2068,7 @@
    HNODEPTR hptr, *pointer;
    FILE     *out_fp;
    char     filename[256];
-   u_long   cnt=a_ctr;
+   u_int64_t   cnt=a_ctr;

    /* generate file name */
    sprintf(filename,"%s/site_%04d%02d.%s",
@@ -2092,7 +2092,7 @@
       if (hptr->flag != OBJ_GRP)
       {
          fprintf(out_fp,
-         "%lu\t%lu\t%.0f\t%lu\t%s\n",
+         "%lld\t%lld\t%.0f\t%lld\t%s\n",
             hptr->count,hptr->files,hptr->xfer/1024,
             hptr->visit,hptr->string);
       }
@@ -2111,7 +2111,7 @@
    UNODEPTR uptr, *pointer;
    FILE     *out_fp;
    char     filename[256];
-   u_long   cnt=a_ctr;
+   u_int64_t   cnt=a_ctr;

    /* generate file name */
    sprintf(filename,"%s/url_%04d%02d.%s",
@@ -2133,7 +2133,7 @@
       uptr=*pointer++;
       if (uptr->flag != OBJ_GRP)
       {
-         fprintf(out_fp,"%lu\t%.0f\t%s\n",
+         fprintf(out_fp,"%lld\t%.0f\t%s\n",
             uptr->count,uptr->xfer/1024,uptr->string);
       }
       cnt--;
@@ -2151,7 +2151,7 @@
    RNODEPTR rptr, *pointer;
    FILE     *out_fp;
    char     filename[256];
-   u_long   cnt=a_ctr;
+   u_int64_t   cnt=a_ctr;

    /* generate file name */
    sprintf(filename,"%s/ref_%04d%02d.%s",
@@ -2173,7 +2173,7 @@
       rptr=*pointer++;
       if (rptr->flag != OBJ_GRP)
       {
-         fprintf(out_fp,"%lu\t%s\n",rptr->count, rptr->string);
+         fprintf(out_fp,"%lld\t%s\n",rptr->count, rptr->string);
       }
       cnt--;
    }
@@ -2212,7 +2212,7 @@
       aptr=*pointer++;
       if (aptr->flag != OBJ_GRP)
       {
-         fprintf(out_fp,"%lu\t%s\n",aptr->count,aptr->string);
+         fprintf(out_fp,"%lld\t%s\n",aptr->count,aptr->string);
       }
       cnt--;
    }
@@ -2229,7 +2229,7 @@
    INODEPTR iptr, *pointer;
    FILE     *out_fp;
    char     filename[256];
-   u_long   cnt=a_ctr;
+   u_int64_t   cnt=a_ctr;

    /* generate file name */
    sprintf(filename,"%s/user_%04d%02d.%s",
@@ -2253,7 +2253,7 @@
       if (iptr->flag != OBJ_GRP)
       {
          fprintf(out_fp,
-         "%lu\t%lu\t%.0f\t%lu\t%s\n",
+         "%lld\t%lld\t%.0f\t%lld\t%s\n",
             iptr->count,iptr->files,iptr->xfer/1024,
             iptr->visit,iptr->string);
       }
@@ -2292,7 +2292,7 @@
    while(cnt)
    {
       sptr=*pointer++;
-      fprintf(out_fp,"%lu\t%s\n",sptr->count,sptr->string);
+      fprintf(out_fp,"%lld\t%s\n",sptr->count,sptr->string);
       cnt--;
    }
    fclose(out_fp);
@@ -2400,25 +2400,25 @@
                      "<FONT SIZE=\"-1\">%s %d</FONT></A></TD>\n",
                       hist_year[s_mth], hist_month[s_mth], html_ext,
                       s_month[hist_month[s_mth]-1], hist_year[s_mth]);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%llu</FONT></TD>\n",
                       hist_hit[s_mth]/days_in_month);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_files[s_mth]/days_in_month);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_page[s_mth]/days_in_month);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_visit[s_mth]/days_in_month);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_site[s_mth]);
       fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%.0f</FONT></TD>\n",
                       hist_xfer[s_mth]);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_visit[s_mth]);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_page[s_mth]);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lld</FONT></TD>\n",
                       hist_files[s_mth]);
-      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%lu</FONT></TD></TR>\n",
+      fprintf(out_fp,"<TD ALIGN=right><FONT SIZE=\"-1\">%llu</FONT></TD></TR>\n",
                       hist_hit[s_mth]);
       gt_hit   += hist_hit[s_mth];
       gt_files += hist_files[s_mth];
@@ -2452,7 +2452,7 @@

 int qs_site_cmph(const void *cp1, const void *cp2)
 {
-   u_long  t1, t2;
+   u_int64_t  t1, t2;
    t1=(*(HNODEPTR *)cp1)->count;
    t2=(*(HNODEPTR *)cp2)->count;
    if (t1!=t2) return (t2<t1)?-1:1;
@@ -2482,7 +2482,7 @@

 int qs_url_cmph(const void *cp1, const void *cp2)
 {
-   u_long  t1, t2;
+   u_int64_t  t1, t2;
    t1=(*(UNODEPTR *)cp1)->count;
    t2=(*(UNODEPTR *)cp2)->count;
    if (t1!=t2) return (t2<t1)?-1:1;
@@ -2542,7 +2542,7 @@

 int qs_ref_cmph(const void *cp1, const void *cp2)
 {
-   u_long  t1, t2;
+   u_int64_t  t1, t2;
    t1=(*(RNODEPTR *)cp1)->count;
    t2=(*(RNODEPTR *)cp2)->count;
    if (t1!=t2) return (t2<t1)?-1:1;
@@ -2557,7 +2557,7 @@

 int qs_agnt_cmph(const void *cp1, const void *cp2)
 {
-   u_long  t1, t2;
+   u_int64_t  t1, t2;
    t1=(*(ANODEPTR *)cp1)->count;
    t2=(*(ANODEPTR *)cp2)->count;
    if (t1!=t2) return (t2<t1)?-1:1;
@@ -2572,7 +2572,7 @@

 int qs_srch_cmph(const void *cp1, const void *cp2)
 {
-   u_long  t1, t2;
+   u_int64_t  t1, t2;
    t1=(*(SNODEPTR *)cp1)->count;
    t2=(*(SNODEPTR *)cp2)->count;
    if (t1!=t2) return (t2<t1)?-1:1;
@@ -2587,7 +2587,7 @@

 int qs_ident_cmph(const void *cp1, const void *cp2)
 {
-   u_long  t1, t2;
+   u_int64_t  t1, t2;
    t1=(*(INODEPTR *)cp1)->count;
    t2=(*(INODEPTR *)cp2)->count;
    if (t1!=t2) return (t2<t1)?-1:1;
@@ -2600,11 +2600,11 @@
 /* LOAD_SITE_ARRAY - load up the sort array  */
 /*********************************************/

-u_long load_site_array(HNODEPTR *pointer)
+u_int64_t load_site_array(HNODEPTR *pointer)
 {
    HNODEPTR hptr;
    int      i;
-   u_long   ctr = 0;
+   u_int64_t   ctr = 0;

    /* load the array */
    for (i=0;i<MAXHASH;i++)
@@ -2624,11 +2624,11 @@
 /* LOAD_URL_ARRAY - load up the sort array   */
 /*********************************************/

-u_long load_url_array(UNODEPTR *pointer)
+u_int64_t load_url_array(UNODEPTR *pointer)
 {
    UNODEPTR uptr;
    int      i;
-   u_long   ctr = 0;
+   u_int64_t   ctr = 0;

    /* load the array */
    for (i=0;i<MAXHASH;i++)
@@ -2648,11 +2648,11 @@
 /* LOAD_REF_ARRAY - load up the sort array   */
 /*********************************************/

-u_long load_ref_array(RNODEPTR *pointer)
+u_int64_t load_ref_array(RNODEPTR *pointer)
 {
    RNODEPTR rptr;
    int      i;
-   u_long   ctr = 0;
+   u_int64_t   ctr = 0;

    /* load the array */
    for (i=0;i<MAXHASH;i++)
@@ -2672,11 +2672,11 @@
 /* LOAD_AGENT_ARRAY - load up the sort array */
 /*********************************************/

-u_long load_agent_array(ANODEPTR *pointer)
+u_int64_t load_agent_array(ANODEPTR *pointer)
 {
    ANODEPTR aptr;
    int      i;
-   u_long   ctr = 0;
+   u_int64_t   ctr = 0;

    /* load the array */
    for (i=0;i<MAXHASH;i++)
@@ -2696,11 +2696,11 @@
 /* LOAD_SRCH_ARRAY - load up the sort array  */
 /*********************************************/

-u_long load_srch_array(SNODEPTR *pointer)
+u_int64_t load_srch_array(SNODEPTR *pointer)
 {
    SNODEPTR sptr;
    int      i;
-   u_long   ctr = 0;
+   u_int64_t   ctr = 0;

    /* load the array */
    for (i=0;i<MAXHASH;i++)
@@ -2720,11 +2720,11 @@
 /* LOAD_IDENT_ARRAY - load up the sort array */
 /*********************************************/

-u_long load_ident_array(INODEPTR *pointer)
+u_int64_t load_ident_array(INODEPTR *pointer)
 {
    INODEPTR iptr;
    int      i;
-   u_long   ctr = 0;
+   u_int64_t   ctr = 0;

    /* load the array */
    for (i=0;i<MAXHASH;i++)
Index: webalizer-2.01.10-vc/preserve.c
===================================================================
--- webalizer-2.01.10-vc.orig/preserve.c	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/preserve.c	2006-10-27 12:34:32.000000000 +0400
@@ -67,12 +67,12 @@

 /* local variables */
 int     hist_month[12], hist_year[12];        /* arrays for monthly total */
-u_long  hist_hit[12];                         /* calculations: used to    */
-u_long  hist_files[12];                       /* produce index.html       */
-u_long  hist_site[12];                        /* these are read and saved */
+u_int64_t  hist_hit[12];                         /* calculations: used to    */
+u_int64_t  hist_files[12];                       /* produce index.html       */
+u_int64_t  hist_site[12];                        /* these are read and saved */
 double  hist_xfer[12];                        /* in the history file      */
-u_long  hist_page[12];
-u_long  hist_visit[12];
+u_int64_t  hist_page[12];
+u_int64_t  hist_visit[12];

 int     hist_fday[12], hist_lday[12];         /* first/last day arrays    */

@@ -110,7 +110,7 @@
          }

          /* month# year# requests files sites xfer firstday lastday */
-         numfields = sscanf(buffer,"%d %d %lu %lu %lu %lf %d %d %lu %lu",
+         numfields = sscanf(buffer,"%d %d %lld %lld %lld %lf %d %d %lld %lld",
                        &hist_month[i],
                        &hist_year[i],
                        &hist_hit[i],
@@ -151,7 +151,7 @@
       {
          if ((hist_month[i] != 0) && (hist_hit[i] != 0))
          {
-            fprintf(hist_fp,"%d %d %lu %lu %lu %.0f %d %d %lu %lu\n",
+            fprintf(hist_fp,"%d %d %lld %lld %lld %.0f %d %d %lld %lld\n",
                             hist_month[i],
                             hist_year[i],
                             hist_hit[i],
@@ -214,20 +214,20 @@
    if (fputs(buffer,fp)==EOF) return 1;  /* error exit */

    /* Monthly totals for sites, urls, etc... */
-   sprintf(buffer,"%lu %lu %lu %lu %lu %lu %.0f %lu %lu %lu\n",
+   sprintf(buffer,"%lld %lld %lld %lld %lld %lld %.0f %lld %lld %lld\n",
         t_hit, t_file, t_site, t_url,
         t_ref, t_agent, t_xfer, t_page, t_visit, t_user);
    if (fputs(buffer,fp)==EOF) return 1;  /* error exit */

    /* Daily totals for sites, urls, etc... */
-   sprintf(buffer,"%lu %lu %lu %d %d\n",
+   sprintf(buffer,"%lld %lld %lld %d %d\n",
         dt_site, ht_hit, mh_hit, f_day, l_day);
    if (fputs(buffer,fp)==EOF) return 1;  /* error exit */

    /* Monthly (by day) total array */
    for (i=0;i<31;i++)
    {
-      sprintf(buffer,"%lu %lu %.0f %lu %lu %lu\n",
+      sprintf(buffer,"%lld %lld %.0f %lld %lld %lld\n",
         tm_hit[i],tm_file[i],tm_xfer[i],tm_site[i],tm_page[i],tm_visit[i]);
       if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
    }
@@ -235,7 +235,7 @@
    /* Daily (by hour) total array */
    for (i=0;i<24;i++)
    {
-      sprintf(buffer,"%lu %lu %.0f %lu\n",
+      sprintf(buffer,"%lld %lld %.0f %lld\n",
         th_hit[i],th_file[i],th_xfer[i],th_page[i]);
       if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
    }
@@ -243,7 +243,7 @@
    /* Response codes */
    for (i=0;i<TOTAL_RC;i++)
    {
-      sprintf(buffer,"%lu\n",response[i].count);
+      sprintf(buffer,"%lld\n",response[i].count);
       if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
    }

@@ -255,7 +255,7 @@
       uptr=um_htab[i];
       while (uptr!=NULL)
       {
-         sprintf(buffer,"%s\n%d %lu %lu %.0f %lu %lu\n", uptr->string,
+         sprintf(buffer,"%s\n%d %lld %lld %.0f %lld %lld\n", uptr->string,
               uptr->flag, uptr->count, uptr->files, uptr->xfer,
               uptr->entry, uptr->exit);
          if (fputs(buffer,fp)==EOF) return 1;
@@ -272,7 +272,7 @@
       hptr=sm_htab[i];
       while (hptr!=NULL)
       {
-         sprintf(buffer,"%s\n%d %lu %lu %.0f %lu %lu\n%s\n",
+         sprintf(buffer,"%s\n%d %lld %lld %.0f %lld %lld\n%s\n",
               hptr->string,
               hptr->flag,
               hptr->count,
@@ -294,7 +294,7 @@
       hptr=sd_htab[i];
       while (hptr!=NULL)
       {
-         sprintf(buffer,"%s\n%d %lu %lu %.0f %lu %lu\n%s\n",
+         sprintf(buffer,"%s\n%d %lld %lld %.0f %lld %lld\n%s\n",
               hptr->string,
               hptr->flag,
               hptr->count,
@@ -318,7 +318,7 @@
          rptr=rm_htab[i];
          while (rptr!=NULL)
          {
-            sprintf(buffer,"%s\n%d %lu\n", rptr->string,
+            sprintf(buffer,"%s\n%d %lld\n", rptr->string,
                  rptr->flag, rptr->count);
             if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
             rptr=rptr->next;
@@ -336,7 +336,7 @@
          aptr=am_htab[i];
          while (aptr!=NULL)
          {
-            sprintf(buffer,"%s\n%d %lu\n", aptr->string,
+            sprintf(buffer,"%s\n%d %lld\n", aptr->string,
                  aptr->flag, aptr->count);
             if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
             aptr=aptr->next;
@@ -352,7 +352,7 @@
       sptr=sr_htab[i];
       while (sptr!=NULL)
       {
-         sprintf(buffer,"%s\n%lu\n", sptr->string,sptr->count);
+         sprintf(buffer,"%s\n%lld\n", sptr->string,sptr->count);
          if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
          sptr=sptr->next;
       }
@@ -367,7 +367,7 @@
       iptr=im_htab[i];
       while (iptr!=NULL)
       {
-         sprintf(buffer,"%s\n%d %lu %lu %.0f %lu %lu\n",
+         sprintf(buffer,"%s\n%d %lld %lld %.0f %lld %lld\n",
               iptr->string,
               iptr->flag,
               iptr->count,
@@ -403,7 +403,7 @@
    char   buffer[BUFSIZE];
    char   tmp_buf[BUFSIZE];

-   u_long ul_bogus=0;
+   u_int64_t ul_bogus=0;

    fp=fopen(state_fname,"r");
    if (fp==NULL)
@@ -437,7 +437,7 @@
    /* Get monthly totals */
    if ((fgets(buffer,BUFSIZE,fp)) != NULL)
    {
-      sscanf(buffer,"%lu %lu %lu %lu %lu %lu %lf %lu %lu %lu",
+      sscanf(buffer,"%lld %lld %lld %lld %lld %lld %lf %lld %lld %lld",
        &t_hit, &t_file, &t_site, &t_url,
        &t_ref, &t_agent, &t_xfer, &t_page, &t_visit, &t_user);
    } else return 3;  /* error exit */
@@ -445,7 +445,7 @@
    /* Get daily totals */
    if ((fgets(buffer,BUFSIZE,fp)) != NULL)
    {
-      sscanf(buffer,"%lu %lu %lu %d %d",
+      sscanf(buffer,"%lld %lld %lld %d %d",
        &dt_site, &ht_hit, &mh_hit, &f_day, &l_day);
    } else return 4;  /* error exit */

@@ -454,7 +454,7 @@
    {
       if ((fgets(buffer,BUFSIZE,fp)) != NULL)
       {
-         sscanf(buffer,"%lu %lu %lf %lu %lu %lu",
+         sscanf(buffer,"%lld %lld %lf %lld %lld %lld",
           &tm_hit[i],&tm_file[i],&tm_xfer[i],&tm_site[i],&tm_page[i],
           &tm_visit[i]);
       } else return 5;  /* error exit */
@@ -465,7 +465,7 @@
    {
       if ((fgets(buffer,BUFSIZE,fp)) != NULL)
       {
-         sscanf(buffer,"%lu %lu %lf %lu",
+         sscanf(buffer,"%lld %lld %lf %lld",
           &th_hit[i],&th_file[i],&th_xfer[i],&th_page[i]);
       } else return 6;  /* error exit */
    }
@@ -474,7 +474,7 @@
    for (i=0;i<TOTAL_RC;i++)
    {
       if ((fgets(buffer,BUFSIZE,fp)) != NULL)
-         sscanf(buffer,"%lu",&response[i].count);
+         sscanf(buffer,"%lld",&response[i].count);
       else return 7;  /* error exit */
    }

@@ -500,7 +500,7 @@
       if (!isdigit((int)buffer[0])) return 10;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%d %lu %lu %lf %lu %lu",
+      sscanf(buffer,"%d %lld %lld %lf %lld %lld",
          &t_unode.flag,&t_unode.count,
          &t_unode.files, &t_unode.xfer,
          &t_unode.entry, &t_unode.exit);
@@ -531,7 +531,7 @@
       if (!isdigit((int)buffer[0])) return 8;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%d %lu %lu %lf %lu %lu",
+      sscanf(buffer,"%d %lld %lld %lf %lld %lld",
          &t_hnode.flag,&t_hnode.count,
          &t_hnode.files, &t_hnode.xfer,
          &t_hnode.visit, &t_hnode.tstamp);
@@ -571,7 +571,7 @@
       if (!isdigit((int)buffer[0])) return 9;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%d %lu %lu %lf %lu %lu",
+      sscanf(buffer,"%d %lld %lld %lf %lld %lld",
           &t_hnode.flag,&t_hnode.count,
           &t_hnode.files, &t_hnode.xfer,
           &t_hnode.visit, &t_hnode.tstamp);
@@ -610,7 +610,7 @@
       if (!isdigit((int)buffer[0])) return 11;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%d %lu",&t_rnode.flag,&t_rnode.count);
+      sscanf(buffer,"%d %lld",&t_rnode.flag,&t_rnode.count);

       /* insert node */
       if (put_rnode(tmp_buf,t_rnode.flag,
@@ -635,7 +635,7 @@
       if (!isdigit((int)buffer[0])) return 12;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%d %lu",&t_anode.flag,&t_anode.count);
+      sscanf(buffer,"%d %lld",&t_anode.flag,&t_anode.count);

       /* insert node */
       if (put_anode(tmp_buf,t_anode.flag,t_anode.count,
@@ -660,7 +660,7 @@
       if (!isdigit((int)buffer[0])) return 13;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%lu",&t_snode.count);
+      sscanf(buffer,"%lld",&t_snode.count);

       /* insert node */
       if (put_snode(tmp_buf,t_snode.count,sr_htab))
@@ -685,7 +685,7 @@
       if (!isdigit((int)buffer[0])) return 14;  /* error exit */

       /* load temporary node data */
-      sscanf(buffer,"%d %lu %lu %lf %lu %lu",
+      sscanf(buffer,"%d %lld %lld %lf %lld %lld",
          &t_inode.flag,&t_inode.count,
          &t_inode.files, &t_inode.xfer,
          &t_inode.visit, &t_inode.tstamp);
Index: webalizer-2.01.10-vc/preserve.h
===================================================================
--- webalizer-2.01.10-vc.orig/preserve.h	2006-10-27 12:34:21.000000000 +0400
+++ webalizer-2.01.10-vc/preserve.h	2006-10-27 12:34:32.000000000 +0400
@@ -8,12 +8,12 @@

 extern int     hist_month[12];                /* arrays for monthly total */
 extern int     hist_year[12];
-extern u_long  hist_hit[12];                  /* calculations: used to    */
-extern u_long  hist_files[12];                /* produce index.html       */
-extern u_long  hist_site[12];                 /* these are read and saved */
+extern u_int64_t  hist_hit[12];               /* calculations: used to    */
+extern u_int64_t  hist_files[12];             /* produce index.html       */
+extern u_int64_t  hist_site[12];              /* these are read and saved */
 extern double  hist_xfer[12];                 /* in the history file      */
-extern u_long  hist_page[12];
-extern u_long  hist_visit[12];
+extern u_int64_t  hist_page[12];
+extern u_int64_t  hist_visit[12];

 extern int     hist_fday[12];                 /* first/last day arrays    */
 extern int     hist_lday[12];
Index: webalizer-2.01.10-vc/webalizer.c
===================================================================
--- webalizer-2.01.10-vc.orig/webalizer.c	2006-10-27 12:34:31.000000000 +0400
+++ webalizer-2.01.10-vc/webalizer.c	2006-10-27 12:34:32.000000000 +0400
@@ -171,30 +171,30 @@
         cur_day=0, cur_hour=0,                /* tracking variables       */
         cur_min=0, cur_sec=0;

-u_long  cur_tstamp=0;                         /* Timestamp...             */
-u_long  rec_tstamp=0;
-u_long  req_tstamp=0;
-u_long  epoch;                                /* used for timestamp adj.  */
+u_int64_t  cur_tstamp=0;                         /* Timestamp...             */
+u_int64_t  rec_tstamp=0;
+u_int64_t  req_tstamp=0;
+u_int64_t  epoch;                                /* used for timestamp adj.  */

 int     check_dup=0;                          /* check for dup flag       */
 int     gz_log=0;                             /* gziped log? (0=no)       */

 double  t_xfer=0.0;                           /* monthly total xfer value */
-u_long  t_hit=0,t_file=0,t_site=0,            /* monthly total vars       */
+u_int64_t  t_hit=0,t_file=0,t_site=0,            /* monthly total vars       */
         t_url=0,t_ref=0,t_agent=0,
         t_page=0, t_visit=0, t_user=0;

 double  tm_xfer[31];                          /* daily transfer totals    */

-u_long  tm_hit[31], tm_file[31],              /* daily total arrays       */
+u_int64_t  tm_hit[31], tm_file[31],              /* daily total arrays       */
         tm_site[31], tm_page[31],
         tm_visit[31];

-u_long  dt_site;                              /* daily 'sites' total      */
+u_int64_t  dt_site;                              /* daily 'sites' total      */

-u_long  ht_hit=0, mh_hit=0;                   /* hourly hits totals       */
+u_int64_t  ht_hit=0, mh_hit=0;                   /* hourly hits totals       */

-u_long  th_hit[24], th_file[24],              /* hourly total arrays      */
+u_int64_t  th_hit[24], th_file[24],              /* hourly total arrays      */
         th_page[24];

 double  th_xfer[24];
@@ -203,7 +203,7 @@

 struct  utsname system_info;                  /* system info structure    */

-u_long  ul_bogus =0;                          /* Dummy counter for groups */
+u_int64_t  ul_bogus =0;                          /* Dummy counter for groups */

 struct  log_struct log_rec;                   /* expanded log storage     */

@@ -245,9 +245,9 @@
    int    rec_year,rec_month=1,rec_day,rec_hour,rec_min,rec_sec;

    int    good_rec    =0;                /* 1 if we had a good record   */
-   u_long total_rec   =0;                /* Total Records Processed     */
-   u_long total_ignore=0;                /* Total Records Ignored       */
-   u_long total_bad   =0;                /* Total Bad Records           */
+   u_int64_t total_rec   =0;                /* Total Records Processed     */
+   u_int64_t total_ignore=0;                /* Total Records Ignored       */
+   u_int64_t total_bad   =0;                /* Total Bad Records           */

    int    max_ctry;                      /* max countries defined       */

@@ -593,7 +593,7 @@
             total_bad++;                /* if a bad date, bump counter      */
             if (verbose)
             {
-               fprintf(stderr,"%s: %s [%lu]",
+               fprintf(stderr,"%s: %s [%lld]",
                  msg_bad_date,log_rec.datetime,total_rec);
                if (debug_mode) fprintf(stderr,":\n%s\n",tmp_buf);
                else fprintf(stderr,"\n");
@@ -893,7 +893,7 @@
          /* if necessary, shrink referrer to fit storage */
          if (strlen(log_rec.refer)>=MAXREFH)
          {
-            if (verbose) fprintf(stderr,"%s [%lu]\n",
+            if (verbose) fprintf(stderr,"%s [%lld]\n",
                 msg_big_ref,total_rec);
             log_rec.refer[MAXREFH-1]='\0';
          }
@@ -901,7 +901,7 @@
          /* if necessary, shrink URL to fit storage */
          if (strlen(log_rec.url)>=MAXURLH)
          {
-            if (verbose) fprintf(stderr,"%s [%lu]\n",
+            if (verbose) fprintf(stderr,"%s [%lld]\n",
                 msg_big_req,total_rec);
             log_rec.url[MAXURLH-1]='\0';
          }
@@ -1083,8 +1083,8 @@
              (log_rec.resp_code==RC_PARTIALCONTENT))
          {
             /* URL hash table */
-            if (put_unode(log_rec.url,OBJ_REG,(u_long)1,
-                log_rec.xfer_size,&t_url,(u_long)0,(u_long)0,um_htab))
+            if (put_unode(log_rec.url,OBJ_REG,(u_int64_t)1,
+                log_rec.xfer_size,&t_url,(u_int64_t)0,(u_int64_t)0,um_htab))
             {
                if (verbose)
                /* Error adding URL node, skipping ... */
@@ -1093,7 +1093,7 @@

             /* ident (username) hash table */
             if (put_inode(log_rec.ident,OBJ_REG,
-                1,(u_long)i,log_rec.xfer_size,&t_user,
+                1,(u_int64_t)i,log_rec.xfer_size,&t_user,
                 0,rec_tstamp,im_htab))
             {
                if (verbose)
@@ -1106,7 +1106,7 @@
          if (ntop_refs)
          {
             if (log_rec.refer[0]!='\0')
-             if (put_rnode(log_rec.refer,OBJ_REG,(u_long)1,&t_ref,rm_htab))
+             if (put_rnode(log_rec.refer,OBJ_REG,(u_int64_t)1,&t_ref,rm_htab))
              {
               if (verbose)
               fprintf(stderr,"%s %s\n", msg_nomem_r, log_rec.refer);
@@ -1115,7 +1115,7 @@

          /* hostname (site) hash table - daily */
          if (put_hnode(log_rec.hostname,OBJ_REG,
-             1,(u_long)i,log_rec.xfer_size,&dt_site,
+             1,(u_int64_t)i,log_rec.xfer_size,&dt_site,
              0,rec_tstamp,"",sd_htab))
          {
             if (verbose)
@@ -1125,7 +1125,7 @@

          /* hostname (site) hash table - monthly */
          if (put_hnode(log_rec.hostname,OBJ_REG,
-             1,(u_long)i,log_rec.xfer_size,&t_site,
+             1,(u_int64_t)i,log_rec.xfer_size,&t_site,
              0,rec_tstamp,"",sm_htab))
          {
             if (verbose)
@@ -1137,7 +1137,7 @@
          if (ntop_agents)
          {
             if (log_rec.agent[0]!='\0')
-             if (put_anode(log_rec.agent,OBJ_REG,(u_long)1,&t_agent,am_htab))
+             if (put_anode(log_rec.agent,OBJ_REG,(u_int64_t)1,&t_agent,am_htab))
              {
               if (verbose)
               fprintf(stderr,"%s %s\n", msg_nomem_a, log_rec.agent);
@@ -1178,8 +1178,8 @@
          /* URL Grouping */
          if ( (cp1=isinglist(group_urls,log_rec.url))!=NULL)
          {
-            if (put_unode(cp1,OBJ_GRP,(u_long)1,log_rec.xfer_size,
-                &ul_bogus,(u_long)0,(u_long)0,um_htab))
+            if (put_unode(cp1,OBJ_GRP,(u_int64_t)1,log_rec.xfer_size,
+                &ul_bogus,(u_int64_t)0,(u_int64_t)0,um_htab))
             {
                if (verbose)
                /* Error adding URL node, skipping ... */
@@ -1190,7 +1190,7 @@
          /* Site Grouping */
          if ( (cp1=isinglist(group_sites,log_rec.hostname))!=NULL)
          {
-            if (put_hnode(cp1,OBJ_GRP,1,(u_long)(log_rec.resp_code==RC_OK)?1:0,
+            if (put_hnode(cp1,OBJ_GRP,1,(u_int64_t)(log_rec.resp_code==RC_OK)?1:0,
                           log_rec.xfer_size,&ul_bogus,
                           0,rec_tstamp,"",sm_htab))
             {
@@ -1208,7 +1208,7 @@
                if (cp1 != NULL)
                {
                   if (put_hnode(cp1,OBJ_GRP,1,
-                      (u_long)(log_rec.resp_code==RC_OK)?1:0,
+                      (u_int64_t)(log_rec.resp_code==RC_OK)?1:0,
                       log_rec.xfer_size,&ul_bogus,
                       0,rec_tstamp,"",sm_htab))
                   {
@@ -1223,7 +1223,7 @@
          /* Referrer Grouping */
          if ( (cp1=isinglist(group_refs,log_rec.refer))!=NULL)
          {
-            if (put_rnode(cp1,OBJ_GRP,(u_long)1,&ul_bogus,rm_htab))
+            if (put_rnode(cp1,OBJ_GRP,(u_int64_t)1,&ul_bogus,rm_htab))
             {
                if (verbose)
                /* Error adding Referrer node, skipping ... */
@@ -1234,7 +1234,7 @@
          /* User Agent Grouping */
          if ( (cp1=isinglist(group_agents,log_rec.agent))!=NULL)
          {
-            if (put_anode(cp1,OBJ_GRP,(u_long)1,&ul_bogus,am_htab))
+            if (put_anode(cp1,OBJ_GRP,(u_int64_t)1,&ul_bogus,am_htab))
             {
                if (verbose)
                /* Error adding User Agent node, skipping ... */
@@ -1245,7 +1245,7 @@
          /* Ident (username) Grouping */
          if ( (cp1=isinglist(group_users,log_rec.ident))!=NULL)
          {
-            if (put_inode(cp1,OBJ_GRP,1,(u_long)(log_rec.resp_code==RC_OK)?1:0,
+            if (put_inode(cp1,OBJ_GRP,1,(u_int64_t)(log_rec.resp_code==RC_OK)?1:0,
                           log_rec.xfer_size,&ul_bogus,
                           0,rec_tstamp,im_htab))
             {
@@ -1276,7 +1276,7 @@
             total_bad++;
             if (verbose)
             {
-               fprintf(stderr,"%s (%lu)",msg_bad_rec,total_rec);
+               fprintf(stderr,"%s (%lld)",msg_bad_rec,total_rec);
                if (debug_mode) fprintf(stderr,":\n%s\n",tmp_buf);
                else fprintf(stderr,"\n");
             }
@@ -1319,14 +1319,14 @@
       end_time = times(&mytms);              /* display timing totals?   */
       if (time_me || (verbose>1))
       {
-         printf("%lu %s ",total_rec, msg_records);
+         printf("%lld %s ",total_rec, msg_records);
          if (total_ignore)
          {
-            printf("(%lu %s",total_ignore,msg_ignored);
-            if (total_bad) printf(", %lu %s) ",total_bad,msg_bad);
+            printf("(%lld %s",total_ignore,msg_ignored);
+            if (total_bad) printf(", %lld %s) ",total_bad,msg_bad);
                else        printf(") ");
          }
-         else if (total_bad) printf("(%lu %s) ",total_bad,msg_bad);
+         else if (total_bad) printf("(%lld %s) ",total_bad,msg_bad);

          /* get processing time (end-start) */
          temp_time = (float)(end_time-start_time)/CLK_TCK;
@@ -1741,10 +1741,10 @@
 /* CTRY_IDX - create unique # from domain    */
 /*********************************************/

-u_long ctry_idx(char *str)
+u_int64_t ctry_idx(char *str)
 {
    int i=strlen(str),j=0;
-   u_long idx=0;
+   u_int64_t idx=0;
    char *cp1=str+i;
    for (;i>0;i--) { idx+=((*--cp1-'a'+1)<<j); j+=5; }
    return idx;
@@ -1843,7 +1843,7 @@
    cp1=cp2;
    while (*cp1!=0) { if ((*cp1<32)||(*cp1==127)) *cp1='_'; cp1++; }

-   if (put_snode(cp2,(u_long)1,sr_htab))
+   if (put_snode(cp2,(u_int64_t)1,sr_htab))
    {
       if (verbose)
       /* Error adding search string node, skipping .... */
@@ -1923,14 +1923,14 @@
 /*                                                               */
 /*****************************************************************/

-u_long jdate( int day, int month, int year )
+u_int64_t jdate( int day, int month, int year )
 {
-   u_long days;                      /* value returned */
+   u_int64_t days;                      /* value returned */
    int mtable[] = {0,31,59,90,120,151,181,212,243,273,304,334};

    /* First, calculate base number including leap and Centenial year stuff */

-   days=(((u_long)year*365)+day+mtable[month-1]+
+   days=(((u_int64_t)year*365)+day+mtable[month-1]+
            ((year+4)/4) - ((year/100)-(year/400)));

    /* now adjust for leap year before March 1st */
Index: webalizer-2.01.10-vc/webalizer.h
===================================================================
--- webalizer-2.01.10-vc.orig/webalizer.h	2006-10-27 12:34:31.000000000 +0400
+++ webalizer-2.01.10-vc/webalizer.h	2006-10-27 12:34:32.000000000 +0400
@@ -125,13 +125,13 @@

 /* Response code structure */
 struct response_code {     char    *desc;         /* response code struct  */
-                         u_long    count; };
+                         u_int64_t    count; };

 /* Country code structure */
-struct	country_code { u_long idx;
+struct	country_code { u_int64_t idx;
                          char *desc;
-                       u_long count;
-                       u_long files;
+                       u_int64_t count;
+                       u_int64_t files;
                        double  xfer; };

 typedef struct country_code *CLISTPTR;
@@ -141,7 +141,7 @@
                        char   datetime[29];        /* raw timestamp        */
                        char   url[MAXURL];         /* raw request field    */
                        int    resp_code;           /* response code        */
-                       u_long xfer_size;           /* xfer size in bytes   */
+                       u_int64_t xfer_size;           /* xfer size in bytes   */
 #ifdef USE_DNS
                        struct in_addr addr;        /* IP address structure */
 #endif  /* USE_DNS */
@@ -219,8 +219,8 @@
 extern int     dump_header  ;                 /* Dump header as first rec */
 extern char    *dump_path   ;                 /* Path for dump files      */

-extern u_long  cur_tstamp;                    /* Current timestamp        */
-extern u_long  epoch;                         /* used for timestamp adj.  */
+extern u_int64_t  cur_tstamp;                    /* Current timestamp        */
+extern u_int64_t  epoch;                         /* used for timestamp adj.  */
 extern int     check_dup;                     /* check for dups flag      */

 extern int     cur_year,cur_month,            /* year/month/day/hour      */
@@ -228,21 +228,21 @@
                cur_min, cur_sec;

 extern double  t_xfer;                        /* monthly total xfer value */
-extern u_long  t_hit, t_file, t_site,         /* monthly total vars       */
+extern u_int64_t  t_hit, t_file, t_site,         /* monthly total vars       */
                t_url, t_ref,  t_agent,
                t_page,t_visit,t_user;

 extern double  tm_xfer[31];                   /* daily transfer totals    */

-extern u_long  tm_hit[31], tm_file[31],       /* daily total arrays       */
+extern u_int64_t  tm_hit[31], tm_file[31],       /* daily total arrays       */
                tm_site[31],tm_page[31],
                tm_visit[31];

-extern u_long  dt_site;                       /* daily 'sites' total      */
+extern u_int64_t  dt_site;                       /* daily 'sites' total      */

-extern u_long  ht_hit,mh_hit;                 /* hourly hits totals       */
+extern u_int64_t  ht_hit,mh_hit;                 /* hourly hits totals       */

-extern u_long  th_hit[24], th_file[24],       /* hourly total arrays      */
+extern u_int64_t  th_hit[24], th_file[24],       /* hourly total arrays      */
                th_page[24];

 extern double  th_xfer[24];
@@ -255,9 +255,9 @@
 /* define our externally visable functions */

 extern char   *cur_time();
-extern u_long ctry_idx(char *);
+extern u_int64_t ctry_idx(char *);
 extern void   init_counters();
 extern int    ispage(char *);
-extern u_long jdate(int,int,int);
+extern u_int64_t jdate(int,int,int);

 #endif  /* _WEBALIZER_H */
